Control/Monad/Logic.hs

```diff
@@ -176,11 +196,14 @@ logic f = LogicT $ \k -> Identity .
 -- >>> observe empty
 -- *** Exception: No answer.
 --
+-- Since 'Logic' is isomorphic to a list, 'observe' is 'head' in disguise.
+--
 observe :: Logic a -> a
 observe lt = runIdentity $ unLogicT lt (const . pure) (error "No answer.")
 
 -------------------------------------------------------------------------
--- | Extracts all results from a 'Logic' computation.
+-- | Extracts all results from a 'Logic' computation, witnessing
+-- a half of isomorphism between 'Logic' and lists.
```

Comment:
For clarity in description v.s. insight, I'd recommend removing the ", witnessing..." here and instead, after the example code, write "The 'observeAll' function witnesses the other half of the isomorphism between 'Logic' and lists, where the first half is described in the 'runLogic' description above."